src/software/amazon/ion/impl/IonReaderTreeSystem.java

```diff
@@ -254,7 +252,7 @@ public IonType getType()
 
     public boolean isInStruct()
     {
-        return (_parent instanceof IonStruct);
+        return getDepth() > 0 && _parent instanceof IonStruct;
```

Comment:
When a DOM value is provided to a reader, shouldn't the reader's public `getDepth()` method restart at zero for that value? It should also be impossible to `stepOut` past that depth.